### PR TITLE
Introduce additional columns to the Transaction list UI

### LINKED ISSUE #7810: Transactions Report - Introduce additional columns to the Transaction list UI
### Description

The current user interface for transactions is lacking certain data that was introduced with the inclusion of additional fields in the [Transactions CSV export](https://github.com/Automattic/woocommerce-payments/issues/7591).

### Acceptance criteria
To synchronize with the columns in the transactions CSV export, we must integrate the following fields into the Transaction list UI:

- Currency - two additional columns (helpful in identifying transactions involving foreign exchange): Customer currency, Deposit currency. The total fee will be in the Deposit currency.
- In the CSV export rename the column from "Amount" to "Amount in Deposit Currency."
- Rename the column from "Source" to "Payment Method" to align with the ID field.
- Clearly specify the timezone for all timestamps as UTC.
- Deposit ID (Typically starting with po_xxxx. This ID is useful for reconciling transactions to the deposit batch).

### PR DESCRIPTION
**Pull request by mordeth**
Fixes #7810

#### Changes proposed in this Pull Request
This pull request introduces additional and updates some of the existing columns on the Transactions list user interface and the CSV export when there are no more than 25 transactions. The new columns include:

- Customer currency and Deposit currency. The total fee will be in the Deposit currency.
- Add Amount in Customer Currency
- Rename the column from "Source" to "Payment Method" to align with the ID field.
- Clearly specify the timezone for all timestamps as UTC.
- Add Deposit ID

New columns added to the transactions list page
Screenshot 2023-11-30 at 11 16 37

New columns added to the transactions list CSV export
Screenshot 2023-11-30 at 11 17 21

New columns added as options to the Transaction list display settings
Screenshot 2023-11-30 at 13 57 59

https://github.com/Automattic/woocommerce-payments/assets/8667118/c643326e-be9c-4ff6-9f8a-9781ca17e77e

#### Testing instructions
**Case 1**
- Go to `Payment` -> `Transactions`.
- Confirm that the column labels are updated, and new columns are added.
- Verify that all details of the new columns are displayed correctly.

**Case 2*
- Go to `Payment` -> `Transactions`.
- Click the three dots button on the top right corner of the table.
- Confirm that the column labels are updated and new columns are added to the list of available fields.
- Toggle the button for the new fields and verify that they are displayed correctly on the transactions list table.

**Case 3*
- Go to Payment -> Transactions.
- Confirm that you have no more than 25 transactions.
- Click the Download button on the top right corner.
- Check that the column labels are updated, and the new columns are included in the exported CSV file.
- Ensure that the details for the new fields are correctly displayed in the CSV.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7813` or branch name `add/7591-missing-columns-export-csv` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/7591-missing-columns-export-csv) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 85079f54ab9dc8454da8d36c7c7cd38ea385046b
- Build time: 2023-12-14 09:20:59 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: changelog/add-7591-missing-columns-export-csv
Comment (anu-rock) [inline changelog/add-7591-missing-columns-export-csv]: I think we should mention the names of the new columns here.

File: client/transactions/list/index.tsx
Comment (anu-rock) [inline client/transactions/list/index.tsx:203]: @mordeth @souravdebnath1986 I wonder if we should rename this to "Deposit Amount" like we did in the server-exported CSV.

File: client/transactions/list/index.tsx
Comment (anu-rock) [inline client/transactions/list/index.tsx]: This may give an unexpected value for zero-decimal currencies. I think we created a utility function earlier to handle such cases, but I cannot find it at the moment.

**Review by anu-rock — COMMENTED** [review]
Looks good overall.

___

#### Review checklist

| Parameter  | Result |
| ------------- | ------------- |
| Approach  | ✅ All good |
| Implementation  | ✅ All good |
| Automated tests  | ✅ All good |
| Naming  | 👍  Almost good (left a comment) |
| Typing  | Not applicable |
| Code comments  | ✅ All good |
| Changelog  | 👍  Almost good (left a comment) |
| Testing instructions  | ✅ All good |
| Demo (video or JN site)  | ✅ All good |
| Manual testing  | ✅ Works as expected |

File: client/transactions/list/index.tsx
Comment (mordeth) [reply client/transactions/list/index.tsx:203]: @anu-rock I intentionally refrained from modifying the `Amount` label to avoid the column expanding too much, which could lead to other columns being displaced and result in an unattractive layout.

How about incorporating a small helper icon and describing the label in a tooltip? What do you think about that?

File: changelog/add-7591-missing-columns-export-csv
Comment (mordeth) [reply changelog/add-7591-missing-columns-export-csv]: Makes sense, updated!

File: client/transactions/list/index.tsx
Comment (mordeth) [reply client/transactions/list/index.tsx]: I couldn't locate such a method within the utils. However, the same division logic is applied in the [`formatAmount` method](https://github.com/Automattic/woocommerce-payments/blob/124c31cc57b20241d0473372fc967a0135d7d4b0/client/transactions/list/index.tsx#L388).

Comment (anu-rock) [reply client/transactions/list/index.tsx]: In that case, it could be a good opportunity to create a helper function to format amount and refactor `formatAmount` to use it too. See the following links for more context:

pepjwh-il-p2#comment-394
https://github.com/Automattic/woocommerce-payments-server/pull/3549

Comment (mordeth) [reply client/transactions/list/index.tsx]: In my opinion, that change falls outside the scope of the current PR. I will create an issue to examine and update all instances where division by 100 is employed. WDYT?

Comment (mordeth) [reply client/transactions/list/index.tsx]: A new issues has been created #7834

Comment (anu-rock) [reply client/transactions/list/index.tsx]: Awesome, thanks! Marking this thread as resolved.

**Review by dpaun1985 — APPROVED** [review]
✅ LGTM

File: client/transactions/list/index.tsx
Comment (anu-rock) [reply client/transactions/list/index.tsx:203]: @mordeth As we discussed during our last sync team chat, I am aligned with your suggestion about using a tooltip. We can check with Design before making a final call.

Comment (mordeth) [reply client/transactions/list/index.tsx:203]: I've incorporated a new tooltip designed for use within the TableCard component to showcase additional descriptions for the heading labels. Kindly review the attached video and screenshots. cc @elizaan36 @anu-rock @souravdebnath1986 

https://github.com/Automattic/woocommerce-payments/assets/8667118/58709ff8-a977-4eb1-843f-bbcc8fee210d

Desktop
Screenshot 2023-12-11 at 16 49 50

Mobile
Screenshot 2023-12-11 at 16 46 21

**Comment by elizaan36** [comment]
hi @mordeth, The tooltips look fine to me however I'm not sure if "Amount in deposit currency" is the most clear label.
A few questions:

- What are the other tooltip labels?
- What columns are visible by default?

There are some UI issues present in the transactions table that need to be looked at, for example the row labels and amounts are not aligned, which makes the overall table harder to read. 

@rogermattic Could you take a look at this and share any recommendations? I think you can click around using the [demo site](https://github.com/Automattic/woocommerce-payments/pull/7813#issuecomment-1833679583).

**Comment by mordeth** [comment]
Thank you for your input, @elizaan36!

The tooltip was implemented to enhance clarity in the listing table header labels without altering them, as adjusting the labels would widen the columns and push the regular visible columns to the right. Tooltips for Amount, Fees, and Net labels have been included to indicate that these values are in Deposit Currency. The reason is confusion may arise from the newly added field 'Amount in Customer Currency,' which is hidden by default.

The default visible columns remain unaffected by this pull request. You can observe them from the screenshot.
image

> There are some UI issues present in the transactions table that need to be looked at, for example the row labels and amounts are not aligned, which makes the overall table harder to read.

I believe that the design improvements for the TableCard component are beyond the scope of this issue. I recommend creating a separate issue to address these enhancements. WDYT?

**Comment by rogermattic** [comment]
Hey @elizaan36 thanks for tagging me here! I'll have a look at the demo site and will provide feedback. 
I like your suggestion @mordeth to create a separate issue for the table improvements. I'll do that! 

Personally, I think these tips are not that useful and possibly make it even more confusing. (That's just my first thought tho) All the amounts (amount, fees, net) are in the same currency, and all the hints say that they are in "deposit currency". 
Why do we need to mention that at all?

Do we have any merchant's feedback stating that the labels are unclear without more explanation?

**Comment by mordeth** [comment]
>Personally, I think these tips are not that useful and possibly make it even more confusing. (That's just my first thought tho) All the amounts (amount, fees, net) are in the same currency, and all the hints say that they are in "deposit currency".
Why do we need to mention that at all?

Thanks for checking it, @rogermattic! In case when only the Amount field is displayed, there's no confusion. However, we introduced a new field, Customer Amount, which is hidden by default, and could potentially cause confusion for merchants if it becomes visible.

Screenshot 2023-12-12 at 15 50 24

**Comment by rogermattic** [comment]
Thanks for the screenshot @mordeth !
It makes more sense now, though... I think 'Customer Amount' as a label is not great neither. Perhaps 'Payment' 
Amount paid' or 'Paid' would work better? Or even 'Original Amount' and we could use a tooltip just for this one (instead of others) saying something along the lines of 'currency paid by a customer'. WDYT?

**Comment by mordeth** [comment]
>It makes more sense now, though... I think 'Customer Amount' as a label is not great neither. Perhaps 'Payment'
Amount paid' or 'Paid' would work better? Or even 'Original Amount' and we could use a tooltip just for this one (instead of others) saying something along the lines of 'currency paid by a customer'. WDYT?

We've already changed the name of the 'Source' field to 'Payment', that value is already in use.

@anu-rock @souravdebnath1986 What do you think about renaming the `Customer Amount` to `Amount Paid`?

**Comment by anu-rock** [comment]
@mordeth 

> What do you think about renaming the Customer Amount to Amount Paid?

"Amount Paid" sounds good to me.

**Comment by souravdebnath1986** [comment]
Thanks all for taking a look and apologies for joining in the conversation late. 
I think `Paid Amount` makes more sense and in the same vein `Customer Currency` -> can be modified to -> `Paid Currency`. 
I would question the use of tooltips to show the meaning of the field at this stage, we don't have any feedback to say that the field names are confusing. Until then, I don't think this is necessary.  cc: @rogermattic @mordeth

**Comment by rogermattic** [comment]
+1 on removing the tooltips as @souravdebnath1986 suggested

Just one slight doubt on the 'Paid Currency' label  – wouldn't it suggest that the column just includes currency, and not the amount paid? 
Also, just to clarify, I was suggesting changing the 'Customer Currency' -> 'Amount Paid' :)

**Comment by mordeth** [comment]
>Also, just to clarify, I was suggesting changing the 'Customer Currency' -> 'Amount Paid' :)

We currently have two concealed columns, namely 'Customer Currency' and 'Customer Amount.' I assume we should rename them as 'Paid Currency' and 'Paid Amount,' is that correct?

Moreover, I will remove the tooltips, and if necessary in the future, we can reintroduce them.

**Comment by mordeth** [comment]
I have removed the tooltips from the labels and updated the labels.
Screenshot 2023-12-13 at 22 23 11

@anu-rock Could you please conduct a final review? I aim to deploy the pull request as soon as possible.

**Review by anu-rock — APPROVED** [review]
LGTM after latest revisions 🎉

**Pull request merged**
